server-side/webSocket.js

```diff
@@ -0,0 +1,40 @@
+import { WebSocketServer } from 'ws';
+import mongoose from 'mongoose';
+import { connectMongo } from './config/db.js'; // Assuming this connects to MongoDB
+import { EventEmitter } from 'events';
+import { getCountUnreadMessages } from './controllers/message.controller.js';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const wss = new WebSocketServer({ port: 8080 });
+
+connectMongo();
+
+const eventEmitter = new EventEmitter(); 
+
+wss.on('connection', (ws) => {
+  console.log('Client connected');
+  ws.on('message', async (message) => {
+    ws.userId = message.toString();
+    ws.send(await getCountUnreadMessages(ws.userId));
```

Comment:
you should have a switch case here with type, was could be used also for other types of events

Reply:
what do you meen can you explain more?
i understand that you meen: not every time i will have userId in the message so send also type variable and its say you the goal and also to do a switch case that i will check what is the variable contain, this you meen?

Reply:
currently then websocket is in used only for showing the notifications, but what will be if in the future we'll need it for another thing?
inside the on message function you should have a switch case something like this:
switch(message.type)
case countUnread:
ws.send(await getCountUnreadMessages(ws.userId));